jecs.luau

```diff
@@ -902,17 +923,26 @@ local function world_add(
 	if from == to then
 		return
 	end
+
+	local idr = world.component_index[id]
+	local idr_hooks = idr.hooks
+	local on_add = idr_hooks.on_add
+
 	if from then
+		if ECS_IS_PAIR(id) and bit32.band(idr.flags, ECS_ID_EXCLUSIVE) ~= 0 then
+			local on_remove = idr_hooks.on_remove
+			if on_remove then
+				on_remove(entity)
+			end
+		end
+
 		entity_move(entity_index, entity, record, to)
```

Comment:
has to be record.archetype not `to` because on_remove can cause the entity to move